source/text/implementation/vss-implementation-utf8_string_handlers.adb

```diff
@@ -590,6 +653,44 @@ package body VSS.Implementation.UTF8_String_Handlers is
       return VSS.Implementation.Strings.Character_Count (Source.Length);
    end Length;
 
+   ----------------
+   -- Reallocate --
+   ----------------
+
+   procedure Reallocate
+     (Data     : in out UTF8_String_Data_Access;
+      Capacity : VSS.Unicode.UTF8_Code_Unit_Count;
+      Size     : VSS.Unicode.UTF8_Code_Unit_Count)
+   is
+      Minimal_Capacity : constant VSS.Unicode.UTF8_Code_Unit_Count :=
+        (if Capacity >= Size
+         then Capacity
+         else (if Data = null
+               then Size
+               else (if Data.Capacity > Size
+                     then Size
```

Comment:
Do we really ignore old `Data.Capacity` here?

Reply:
Yes. This Capacity is actually allocated previously amount of memory, while Size is requested one.